Source/WebCore/platform/graphics/egl/GLFence.cpp

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2024 Igalia S.L.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free
+ *  Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ *  Boston, MA 02110-1301 USA
+ */
+
+#include "config.h"
+#include "GLFence.h"
+
+#include "GLContext.h"
+
+#if USE(LIBEPOXY)
+#include <epoxy/gl.h>
+#else
+#include <GLES2/gl2.h>
+#endif
+
+namespace WebCore {
+
+std::unique_ptr<GLFence> GLFence::create()
+{
+    auto* context = GLContext::current();
+    if (!context)
+        return nullptr;
+
+    const auto& extensions = context->glExtensions();
+    if (extensions.APPLE_sync) {
```

Comment:
Fences should be available on GLES3 contexts, or on GLES2 contexts that have GL_APPLE_sync extensions.
This should be something like
if (context->version() >= 300 || extensions.APPLE_sync)